interp/type.go

```diff
@@ -835,17 +835,19 @@ func (t *itype) methods() methodSet {
 			res[m.Name] = m.Type.String()
 		}
 	case ptrT:
-		// Consider only methods where receiver is a pointer to type t
-		for _, m := range t.val.method {
-			if m.child[0].child[0].lastChild().typ.cat == ptrT {
-				res[m.ident] = m.typ.TypeOf().String()
-			}
+		for k, v := range t.val.methods() {
+			res[k] = v
 		}
-	default:
-		for _, m := range t.method {
-			res[m.ident] = m.typ.TypeOf().String()
+	case structT:
+		for _, f := range t.field {
+			for k, v := range f.typ.methods() {
+				res[k] = v
+			}
 		}
```

Comment:
```suggestion
		}
		for _, m := range t.method {
			res[m.ident] = m.typ.TypeOf().String()
		}		
```

Reply:
Declined, after analysis, it would not return the correct set of methods for an arbitrary type, see comment on next.